### PR TITLE
Add a /ready endpoint to promtail

### PR DESCRIPTION
**Pull request by steven-sheehy**
Currently promtail has a liveness probe that queries the /metrics endpoint instead of a proper /ready endpoint that is specific to probes. This PR adds an endpoint patterned after Loki's /ready endpoint. Current logic is promtail is ready if it is tailing at least one file. If it's not then something is wrong with permissions, volumes, scrape_config, etc. and it should retry. This will let the user know instantly that promtail is not working instead of not finding out until Grafana says can't connect to datasource. I'm open to feedback if there's a better way to calculate readiness.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -185,6 +185,7 @@ helm:
 		helm dependency build $$chart; \
 		helm package $$chart; \
 	done
+	rm -f production/helm/*/requirements.lock
```
Comment (daixiang0) [inline Makefile]: seems not related to topic, i recommend do one thing in one PR.

**Comment by steven-sheehy** [comment]
@slim-bean I've removed the liveness probes. The helm chart is still configurable by the user if they want to add it back.

File: tools/promtail.sh [hunk 2]
```diff
@@ -236,8 +236,14 @@ spec:
             fieldRef:
               fieldPath: spec.nodeName
         image: grafana/promtail:latest
-        imagePullPolicy: IfNotPresent
+        imagePullPolicy: Always
```
Comment (slim-bean) [inline tools/promtail.sh]: @DanCech I know this script is used by grafana.com, just want to check if you had any particular reason for setting `IfNotPresent` vs `Always`

Comment (DanCech) [reply tools/promtail.sh]: Nope, it was just generated from the ksonnet output.  Given that it's using `:latest`, `Always` makes much more sense.
